Sources/VideoOptionsControlBar.swift

```diff
@@ -9,62 +9,98 @@
  * Refer to the COPYING file of the official project for license.
  *****************************************************************************/
 
-@objc (VideoOptionsControlBarDelegate)
+@objc (VLCVideoOptionsControlBarDelegate)
 protocol VideoOptionsControlBarDelegate: class {
     func didToggleFullScreen(_ optionsBar: VideoOptionsControlBar)
     func didToggleRepeat(_ optionsBar: VideoOptionsControlBar)
     func didSelectSubtitle(_ optionsBar: VideoOptionsControlBar)
     func didSelectMoreOptions(_ optionsBar: VideoOptionsControlBar)
-    func didToggleOrientationLock(_ optionsBar: VideoOptionsControlBar)
+    func didToggleInterfaceLock(_ optionsBar: VideoOptionsControlBar)
 }
 
 @objc (VLCVideoOptionsControlBar)
 @objcMembers class VideoOptionsControlBar: UIStackView {
     
     // MARK: Instance variables
     weak var delegate: VideoOptionsControlBarDelegate?
+    private var rptMode: VLCRepeatMode = .doNotRepeat
+    
+    var orientationAxis: NSLayoutConstraint.Axis {
+        set {
+            // rotate the control bar's orientation by switching it's height and width values and changing it's layout positioning
+            axis = newValue
+            let minX: CGFloat = frame.maxX - frame.size.height
+            let minY: CGFloat = frame.maxY - frame.size.width
+            frame = CGRect(x: minX, y: minY, width: frame.size.height, height:frame.size.width)
+        }
+        
+        get {
+            return axis
+        }
+    }
+    
+    var repeatMode: VLCRepeatMode {
+        set {
+            rptMode = newValue
+            switch newValue {
+            case .repeatCurrentItem:
+                repeatButton.setImage(UIImage(named: "repeatOne-new"), for: .normal)
+                break
+            case .repeatAllItems:
+                repeatButton.setImage(UIImage(named: "repeat-new"), for: .normal)
+                break
+            default: // no repeat
+                repeatButton.setImage(UIImage(named: "no-repeat-new"), for: .normal)
+                break
+            }
+        }
+        
+        get {
+            return rptMode
+        }
```

Comment:
you don't need the get and don't need rptMode = newValue
should work without it

Reply:
since it's a Computed property , I can't build it without the getter

Reply:
use didset

Reply:
thanks for the wisdom yet again!